core/src/main/java/org/apache/gravitino/storage/relational/service/ModelVersionMetaService.java

```diff
@@ -0,0 +1,288 @@
+      return Collections.emptyList();
+    }
+
+    // Get the aliases for all the model versions.
+    List<ModelVersionAliasRelPO> aliasRelPOs =
+        SessionUtils.getWithoutCommit(
+            ModelVersionAliasRelMapper.class,
+            mapper -> mapper.selectModelVersionAliasRelByModelId(modelEntity.id()));
+    Multimap<Integer, ModelVersionAliasRelPO> aliasRelPOsByModelVersion =
+        ArrayListMultimap.create();
+    aliasRelPOs.forEach(r -> aliasRelPOsByModelVersion.put(r.getModelVersion(), r));
+
+    return modelVersionPOs.stream()
+        .map(
+            m -> {
+              List<ModelVersionAliasRelPO> versionAliasRelPOs =
+                  Lists.newArrayList(aliasRelPOsByModelVersion.get(m.getModelVersion()));
+              return POConverters.fromModelVersionPO(modelIdent, m, versionAliasRelPOs);
+            })
+        .collect(Collectors.toList());
+  }
+
+  public ModelVersionEntity getModelVersionByIdentifier(NameIdentifier ident) {
+    NameIdentifierUtil.checkModelVersion(ident);
+
+    NameIdentifier modelIdent = NameIdentifier.of(ident.namespace().levels());
+    // Will throw a NoSuchEntityException if the model does not exist.
+    ModelEntity modelEntity = ModelMetaService.getInstance().getModelByIdentifier(modelIdent);
+
+    Integer modelVersion;
+    try {
+      modelVersion = Integer.valueOf(ident.name());
+    } catch (NumberFormatException e) {
+      modelVersion = null;
+    }
+
+    final Integer copiedModelVersion = modelVersion;
+    ModelVersionPO modelVersionPO =
+        SessionUtils.getWithoutCommit(
+            ModelVersionMetaMapper.class,
+            mapper -> {
+              if (copiedModelVersion != null) {
+                return mapper.selectModelVersionMeta(modelEntity.id(), copiedModelVersion);
+              } else {
+                return mapper.selectModelVersionMetaByAlias(modelEntity.id(), ident.name());
+              }
+            });
+
+    if (modelVersionPO == null) {
+      throw new NoSuchEntityException(
+          NoSuchEntityException.NO_SUCH_ENTITY_MESSAGE,
+          Entity.EntityType.MODEL_VERSION.name().toLowerCase(Locale.ROOT),
+          ident.toString());
+    }
+
+    List<ModelVersionAliasRelPO> aliasRelPOs =
+        SessionUtils.getWithoutCommit(
+            ModelVersionAliasRelMapper.class,
+            mapper -> {
+              if (copiedModelVersion != null) {
+                return mapper.selectModelVersionAliasRelByModelIdAndVersion(
+                    modelEntity.id(), copiedModelVersion);
+              } else {
+                return mapper.selectModelVersionAliasRelByModelIdAndAlias(
+                    modelEntity.id(), ident.name());
+              }
+            });
+
+    return POConverters.fromModelVersionPO(modelIdent, modelVersionPO, aliasRelPOs);
+  }
+
+  public void insertModelVersion(ModelVersionEntity modelVersionEntity, boolean overwrite)
+      throws IOException {
+    if (overwrite) {
```

Comment:
How about do not pass the `overwrite` param here, only pass the model version entity in the JDBCBackend.

Reply:
I checked JDBCBackend's implementation, it will pass in the `overwrite` flag. I can omit the flag at that place.

Reply:
Yes, I think if we don't need `overwrite` param, we could discard it at the upper level.

Reply:
Since we will omit the overwrite parameters, what if there exists a model version that is the same as the inserted one. According to the table constraint `UNIQUE KEY `uk_mid_ver_del` (`model_id`, `version`, `deleted_at`),`, a model version with the same `model_id`, `version`, `deleted_at` at the same time will throw exceptions.

Reply:
the version is auto-increased every time and the user cannot specify it manually. So we will not have `overwrite` scenario, and there's no conflict key happens.